src/core/cubic.c

```diff
@@ -460,9 +458,14 @@ CubicCongestionControlOnDataAcknowledged(
         //
 
         int64_t DeltaT =
-            (int64_t)TimeInCongAvoid -
-            (int64_t)Cubic->KCubic +
-            (int64_t)US_TO_MS(AckEvent->SmoothedRtt);
+            US_TO_MS(
+                (int64_t)TimeInCongAvoidUs -
+                (int64_t)MS_TO_US(Cubic->KCubic) +
+                (int64_t)AckEvent->SmoothedRtt
+            );
+        if (DeltaT > 2500000) {
```

Comment:
It seems like you've refined the overflow check, but where did the number 2.5M come from?

Reply:
I grabbed this number from the comment above (which I believe you wrote?), and did some math to validate that the number did seem to work. I'm still not particularly happy with it though. Also, with the limit here, I'm not sure we need the one below, checking for a negative `int64_t` any more. Going even further, the code at the bottom here limits cwnd to 2 * max_bytes_in_flight, which I think we might be able to be smarter about and elide more calculations if threshold is over a limit.